crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -490,6 +503,40 @@ fn stub_file_empty_lines(
     }
 }
 
+/// Checks if an empty line should be inserted between the preceding and, optionally,
+/// the following node according to the [`blank_line_after_nested_stub_class`](https://github.com/astral-sh/ruff/issues/8891)
+/// preview style.
+///
+/// If `following` is `None`, then the preceding node is the last node in the suite.
+pub(crate) fn blank_line_after_nested_stub_class_condition(
+    preceding: AnyNodeRef<'_>,
+    following: Option<AnyNodeRef<'_>>,
+    f: &PyFormatter,
+) -> bool {
+    let comments = f.context().comments();
+    match preceding.as_stmt_class_def() {
+        Some(class) if contains_only_an_ellipsis(&class.body, comments) => {
+            !class.decorator_list.is_empty()
+                || match following {
+                    Some(AnyNodeRef::StmtClassDef(ast::StmtClassDef {
+                        body,
+                        decorator_list,
+                        ..
+                    })) => !contains_only_an_ellipsis(body, comments) || !decorator_list.is_empty(),
+                    Some(AnyNodeRef::StmtFunctionDef(_)) | None => true,
+                    _ => false,
+                }
```

Comment:
Nit: A match in a binary expression feels very heavy. Some temporary variables could also help to document the conditions. 
```suggestion
            let has_decorators = !class.decorator_list.is_empty();
            let followed_by_class_or_function = match following {
                Some(AnyNodeRef::StmtClassDef(ast::StmtClassDef {
                    body,
                    decorator_list,
                    ..
                })) => !contains_only_an_ellipsis(body, comments) || !decorator_list.is_empty(),
                Some(AnyNodeRef::StmtFunctionDef(_)) | None => true,
                _ => false,
            };

            has_decorators || followed_by_class_or_function
```

I think it would be helpful to add some comments here explaining why the conditions are as they are.

Reply:
Thanks for the suggestion. I've added some comments with examples.